### PR TITLE
Add Translation for MathF (#18842)

### LINKED ISSUE #18842: Translate MathF members to SQL functions
MathF was added in .NET Standard 2.1. Existing translations for `Math` can be used for MathF.

### PR DESCRIPTION
**Pull request by ntovas**
Fixes #18842

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/efcore?pullRequest=24829) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/efcore?pullRequest=24829) before we can accept your contribution.<br/><br/>:x: ntovas [sign now](https://cla.dotnetfoundation.org/dotnet/efcore?pullRequest=24829)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/efcore?pullRequest=24829) it.</sub>

**Comment by smitpatel** [comment]
- The reflection is going wrong in those types causing errors because methods are not found.
- You will need to add a test for each new method translation you are adding.

File: src/EFCore.SqlServer/Query/Internal/SqlServerMathTranslator.cs [hunk 1]
```diff
@@ -52,21 +52,58 @@ public class SqlServerMathTranslator : IMethodCallTranslator
             { typeof(Math).GetRequiredRuntimeMethod(nameof(Math.Sign), new[] { typeof(int) }), "SIGN" },
             { typeof(Math).GetRequiredRuntimeMethod(nameof(Math.Sign), new[] { typeof(long) }), "SIGN" },
             { typeof(Math).GetRequiredRuntimeMethod(nameof(Math.Sign), new[] { typeof(sbyte) }), "SIGN" },
-            { typeof(Math).GetRequiredRuntimeMethod(nameof(Math.Sign), new[] { typeof(short) }), "SIGN" }
+            { typeof(Math).GetRequiredRuntimeMethod(nameof(Math.Sign), new[] { typeof(short) }), "SIGN" },
+            { typeof(MathF).GetRequiredRuntimeMethod(nameof(MathF.Abs), new[] { typeof(decimal) }), "ABS" },
+            { typeof(MathF).GetRequiredRuntimeMethod(nameof(MathF.Abs), new[] { typeof(double) }), "ABS" },
```
Comment (bricelam) [inline src/EFCore.SqlServer/Query/Internal/SqlServerMathTranslator.cs:57]: `decimal` and `double` overloads don't exist on this type (only `float`)

**Review by bricelam — CHANGES_REQUESTED** [review]

**Comment by bricelam** [comment]
@smitpatel Should we translate `Math.PI` to `PI()`? (might be a bit silly for a constant...)

**Comment by smitpatel** [comment]
If it is easy, sure.

**Comment by ntovas** [comment]
@smitpatel hey, I couldn't find where are the tests for this functionality.

**Comment by smitpatel** [comment]
https://github.com/dotnet/efcore/blob/main/test/EFCore.Specification.Tests/Query/NorthwindFunctionsQueryTestBase.cs

**Comment by bricelam** [comment]
> If it is easy, sure.

Needs a new *member* translator, but it should be pretty easy. Maybe I'll tack it on when merging this PR... (unless @ntovas wants to give it a go first)

**Comment by bricelam** [comment]
...we should probably file an issue to translate the [Cosmos math functions](https://docs.microsoft.com/en-us/azure/cosmos-db/sql-query-mathematical-functions) too...

**Comment by roji** [comment]
@bricelam there's #24463 waiting for @maumar in case you're in the math mood.

**Comment by ntovas** [comment]
@bricelam I would like to give to try it, but I will need a bit more context, specially on *member translator*

Also I could make the same fix for CosmosDb

**Comment by bricelam** [comment]
Method translators: https://github.com/dotnet/efcore/blob/main/src/EFCore.SqlServer/Query/Internal/SqlServerMethodCallTranslatorProvider.cs
Member translators: https://github.com/dotnet/efcore/blob/main/src/EFCore.SqlServer/Query/Internal/SqlServerMemberTranslatorProvider.cs

**Comment by smitpatel** [comment]
Cosmos tests are failing.

**Comment by ntovas** [comment]
> Cosmos tests are failing.

I will look at them and will add any changes, most likely on Saturday, if it's ok for you.

**Pull request merged**